### PR TITLE
Converted to mod, using jsonpb to marshal stub responses and fixed tests

### PR DESCRIPTION
**Pull request by HairyMike**
I've updated the master branch on my fork with these changes: 
https://github.com/hairymike/gripmock

And created a docker image:
https://hub.docker.com/r/hairymike/gripmock
```
docker pull hairymike/gripmock
```

First of all, great work on `gripmock`!

It would be great to start upgrading some of the other libraries in gripmock.  This PR would lay some of the groundwork for this by first changing over `go modules`, and replacing `packr2` with `pkger`. 

There was a problem running `go mod init` when using `packr2`, so I went ahead and migrated it.

Included in this PR:
* switched over to go modules
* replaced packr2 with pkger
* fixes to address the `well_known_type` test failure.  The cause of the issue was that the keyword `type` was making it into an import alias in the `server.tmpl` file.
* migrated from depreciated protoc plugin lib to `protogen`
* use jsonpb to marshal stubs to response messages

Few issues that this might resolve:
https://github.com/tokopedia/gripmock/issues/61
https://github.com/tokopedia/gripmock/issues/58
https://github.com/tokopedia/gripmock/issues/51

### CODE REVIEW

**Comment by jekiapp** [comment]
Thankyou for this well written PR :+1: . Sorry for the late response, I will try to find a time this week to detailed check of this PR.

File: protoc-gen-gripmock/generator.go [hunk 1]
```diff
@@ -10,52 +10,67 @@ import (
...
+		log.Fatalf("error initializing plugin: %v", err)
+	}
+
+	protos := make([]*descriptor.FileDescriptorProto, len(plugin.Files))
+	for index, file := range plugin.Files {
+		protos[index] = file.Proto
 	}
 
-	gen.CommandLineParameters(gen.Request.GetParameter())
+	params := make(map[string]string)
```
Comment (jekiapp) [inline protoc-gen-gripmock/generator.go:46]: previously we have this one line `gen.CommandLineParameters(gen.Request.GetParameter())` and now we had to manually parse the parameter, care to elaborate?

**Review by jekiapp — COMMENTED** [review]
In the `main()` @generator.go, what you want to introduce there?
what is the benefit of using `pluginpb.codeGeneratorRequest` instead of `generator.Request`?

does the old lib is deprecated? because the updated code seems less simplistic than previous one.

File: protoc-gen-gripmock/generator.go (hunk 1 above)
Comment (HairyMike) [reply protoc-gen-gripmock/generator.go:46]: I agree,

The reason for the change is that the original generator lib is deprecated, and is not covered by the protocol buffer compatibility guarantee anymore (more details [here](https://github.com/golang/protobuf/blob/master/protoc-gen-go/generator/generator.go#L5)).

I think there may have been some issue using generator with protogen as well (I can't remember).  I'm happy to have a look at backing this part out if its a deal breaker

Thanks for having a look at my PR :)

https://github.com/golang/protobuf/blob/master/protoc-gen-go/generator/generator.go#L5

Comment (jekiapp) [reply protoc-gen-gripmock/generator.go:46]: Thankyou for your detailed explanation. PR accepted :)

**Pull request merged**